### PR TITLE
Saas 14.3 mail perf nse

### PR DESCRIPTION
**Pull request by nseinlet**
Improve mail followers SQL query performance.

### CODE REVIEW

File: addons/mail/models/mail_followers.py [hunk 1]
```diff
@@ -105,33 +105,39 @@ def _get_recipient_data(self, records, message_type, subtype_id, pids=None):
...
+           partner.active as active,
+           partner.partner_share as pshare,
+           users.notification_type AS notif,
+           array_agg(groups_rel.gid) AS groups
+      FROM res_partner partner
+ LEFT JOIN res_users users ON users.partner_id = partner.id
+                          AND users.active
+ LEFT JOIN res_groups_users_rel groups_rel ON groups_rel.uid = users.id
+      JOIN sub_followers ON sub_followers.partner_id = partner.id
+                        AND (sub_followers.internal <> TRUE OR coalesce(partner.partner_share, false) <> TRUE)
```
Comment (KangOl) [inline addons/mail/models/mail_followers.py:134]: We can avoid the `OR` by inverting the condition (at the cost of readability)
```suggestion
                        AND NOT (sub_followers.internal AND coalesce(partner.partner_share, false))
```

**Comment by KangOl** [comment]
Can we have some numbers?

**Comment by tde-banana-odoo** [comment]
I propose 19 .

**Comment by nseinlet** [comment]
The starting point of this patch was the submit of forms on the website. Apply for a job, without patch:
```
"POST /website_form/hr.applicant HTTP/1.0" 200 - 182 2.649 0.106
```
Same form, same db, with this patch:
```
"POST /website_form/hr.applicant HTTP/1.0" 200 - 185 0.492 0.107
```

I can have more extensive tests and query plans if required.

**Comment by guewen** [comment]
Work has been done for 12.0 here:  https://github.com/odoo/odoo/pull/64746

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**